### PR TITLE
Adapt Devspaces-Code to use npm for build

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Move Devspaces-Code npm build.
Upstream changes: https://github.com/che-incubator/che-code/pull/420

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

**Comment by RomanNikitenko** [comment]
I've got a successful build for `x86_64` platform on brew, 
I'm testing the same changes for `s390x` and `ppc64le` platforms...

File: devspaces-code/build/dockerfiles/brew.Dockerfile [hunk 1]
```diff
@@ -95,115 +94,117 @@ RUN for vsixfile in /tmp/builtInExtensions/*; do \
...
-    && cd $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/extensions/vscode-test-resolver && yarn \
-    && cd $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/remote && yarn \
-    && cd $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/remote/web && yarn \
-    && cd $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/.vscode/extensions/vscode-selfhost-test-provider && yarn 
-# end of module list generated by sync.sh
-
-# hadolint ignore=SC3045
-RUN echo "$(ulimit -a)"
+RUN echo 'target="30.5.1"' >> $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/.npmrc; \
+    echo 'ms_build_id="10306386"' >> $REMOTE_SOURCES_DIR/devspaces-images-code/app/devspaces-code/code/.npmrc; \
```
Comment (SDawley) [inline devspaces-code/build/dockerfiles/brew.Dockerfile:116]: @RomanNikitenko How often will the build id etc need to be updated?
Will that have to be done manually?

Comment (RomanNikitenko) [reply devspaces-code/build/dockerfiles/brew.Dockerfile:116]: If I'm not mistaken, it's not something that was added within the current PR
it's present in the `devspaces-3-rhel-8`: https://github.com/redhat-developer/devspaces-images/blob/067e59c76648864cd0f88483a7a1f0e48aa218d6/devspaces-code/build/dockerfiles/brew.Dockerfile#L122

Comment (SDawley) [reply devspaces-code/build/dockerfiles/brew.Dockerfile:116]: Ahh so it is. lgtm then

**Review by SDawley — APPROVED** [review]

**Pull request merged**